bundles/org.pitest.pitclipse.ui/src/org/pitest/pitclipse/ui/view/PitView.java

```diff
@@ -44,17 +84,38 @@ public synchronized void createPartControl(Composite parent) {
         }
     }
 
+    /**
+     * Set URL to blank and reset homeUrl, to have a clean browser after
+     * creation<br>
+     * <b>Needed</b> for testing purposes
+     */
+    private void resetBrowser() {
+        browser.setUrl(BLANK_PAGE);
+        homeUrlString = null;
+    }
+
     @Override
     public void setFocus() {
-        // Nothing to do
+        if (browser != null && !browser.isDisposed()) {
+            browser.setFocus();
+        }
     }
 
     @Override
     public synchronized void update(File result) {
         if (result == null) {
-            browser.setText("<html/>");
+            resetBrowser();
         } else {
-            browser.setUrl(result.toURI().toString());
+            homeUrlString = result.toURI().toString();
+            browser.setUrl(homeUrlString);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (browser != null && !browser.isDisposed()) {
+            browser.dispose();
```

Comment:
similar to the previous comment, but this time `browser.dispose` does not seem to be covered by tests... I wonder why...